NavalCombat/NavalCombat2/autogamer.cpp

```diff
@@ -11,15 +11,15 @@ cell AutoGamer::generate_shoot(AutoGamer& human_gamer)
 	set<cell> shot;
 
 	while (shot.size() != BSIZE * BSIZE)
-	{	
+	{
 		cell c = get_random_cell();
 		shot.insert(c);
 
 		if (impossible_shoots.count(c) == 1) continue;
 		// туда куда не стрелял и для какой клетки не выяснил невозможность
-		
+
 		if (human_gamer.prev_shoot_result(c) != UNKNOWN) continue;
-		return c;		
+		return c;
 	}
```

Comment:
Насколько я вижу, в том случае, если мы не смогли найти клетку, куда выстрелить - не делается никакого возврата значения. Это проблема. Можно возвращать, например, `INVALID_CELL`:
```suggestion
	}
	return INVALID_CELL;
```

Что ещё важно - нужно проверять при вызове `generate_shoot()`, не будет ли возвращёт `INVALID_CELL`.